### PR TITLE
feat: filter attributes

### PR DESCRIPTION
**Pull request by vanminh1701**

### CODE REVIEW

File: core/sdk/src/api/backend/OrderAPI.ts [hunk 1]
```diff
@@ -47,17 +65,24 @@ export async function fetchOrdersByStoreId(
...
+      status: 0,
+      walletAddress: sellerAddress,
+      candyShopAddress
+    })}`;
   }
 
   let queryString = qs.stringify(queryObject);
   queryString += filterString;
-
+  console.log('queryString ', queryString);
```
Comment (beepsaidblue) [inline core/sdk/src/api/backend/OrderAPI.ts:85]: can remove this one

File: core/ui/src/index.less [hunk 2]
```diff
@@ -410,3 +410,13 @@
     }
   }
 }
+
+.candy-one-line {
```
Comment (beepsaidblue) [inline core/ui/src/index.less:414]: we already have this one:
```
.candy-line-limit-1 {
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  word-break: break-word;
}
```

File: core/ui/src/model.ts [hunk 3]
```diff
@@ -8,3 +8,13 @@ export interface ShopExchangeInfo {
   symbol: string;
   decimals: number;
 }
+
+export interface CollectionFilter {
+  name: string;
+  identifier: number | Array<number>;
+  attribute?: { [key: string]: string };
+}
+export interface ShopFilter {
```
Comment (beepsaidblue) [inline core/ui/src/model.ts:17]: extra space

File: core/ui/src/public/Orders/index.tsx [hunk 4]
```diff
@@ -9,17 +9,19 @@ import { OrdersActionsStatus } from 'constant';
...
+  wallet?: AnchorWallet;
   url?: string;
   identifiers?: number[];
-  filters?: Array<{ name: string; identifier: number | Array<number> }>;
-  style?: { [key: string]: string | number } | undefined;
-  defaultFilterName?: string;
+  filters?: CollectionFilter[];
+  defaultCollectionFilter?: CollectionFilter;
+  shopFilters?: ShopFilter[];
+  defaultShopFilter?: ShopFilter;
```
Comment (beepsaidblue) [inline core/ui/src/public/Orders/index.tsx:23]: is it possible to merge default shop filter and default collection filter into one ... as there would only be one default

**Comment by haVincy** [comment]
Make the commit message more descriptive on what this PR will modify/change in next push.

**Comment by haVincy** [comment]
Test link - https://feat-filter-attributes.d22tqykss69onq.amplifyapp.com/

File: core/ui/src/components/Dropdown/index.tsx [hunk 5]
```diff
@@ -14,34 +14,47 @@ interface DropdownProps {
...
+          </>
         ) : (
-          <div className="candy-dropdown-icon">
-            <IconChevronDown />
-          </div>
+          <>
+            <div>{placeholder}</div>
+            <div className="candy-dropdown-icon candy-dropdown-icon-placeholder">
+              {isMenuOpen ? <IconChevronUp /> : <IconChevronDown />}
+            </div>
```
Comment (haVincy) [inline core/ui/src/components/Dropdown/index.tsx:56]: These 2 `<div>` are similar only the className is different `candy-dropdown-icon-placeholder`.  Could we just extract it outside from the ternary operator?

```tsx
{currentlySelectedItem?.label ? () :() }
<div className={`${candyDropDownIconStyle}`}>
  {isMenuOpen ? <IconChevronUp /> : <IconChevronDown />}
</div>
```

File: core/ui/src/components/Dropdown/index.tsx [hunk 6]
```diff
@@ -14,34 +14,47 @@ interface DropdownProps {
...
     <div className="candy-dropdown" ref={dropdownRef} onClick={() => setIsMenuOpen((isOpen) => !isOpen)}>
-      <div className="candy-dropdown-flex">
-        <div className={`candy-dropdown-label candy-dropdown-label--${isMenuOpen ? 'purple' : 'black'}`}>
-          {currentlySelectedItem?.label}
-        </div>
-        {isMenuOpen ? (
-          <div className="candy-dropdown-icon">
-            <IconChevronUp />
-          </div>
+      <div className={`candy-dropdown-flex ${currentlySelectedItem?.label ? '' : 'candy-dropdown-flex-placeholder'}`}>
```
Comment (haVincy) [inline core/ui/src/components/Dropdown/index.tsx:38]: It seems we need `currentlySelectedItem?.label` not in once place Maybe just do following 

```ts
let candyDropDownStyle = 'candy-dropdown-flex';
let candyDropDownIconStyle = 'candy-dropdown-icon';

if (!currentlySelectedItem?.label) {
   candyDropDownStyle += ' candy-dropdown-flex-placeholder';
   candyDropDownIconStyle +=  ' candy-dropdown-icon-placeholder';
}

<div className={`${candyDropDownStyle}`}>
```

File: core/ui/src/public/Orders/index.tsx [hunk 7]
```diff
@@ -216,3 +243,45 @@ export const Orders: React.FC<OrdersProps> = ({
...
+//   },
+//   {
+//     name: 'Accessories',
+//     options: [
+//       { label: '18k Gold', value: 0 },
+//       { label: '18k Silver', value: 1 }
+//     ],
+//     placeholder: 'Select Material'
+//   }
+// ];
```
Comment (haVincy) [inline core/ui/src/public/Orders/index.tsx:287]: We should move it out, put it to `ui/src/constant/Orders` but not using it.

File: core/ui/src/public/Orders/index.tsx [hunk 8]
```diff
@@ -156,39 +153,68 @@ export const Orders: React.FC<OrdersProps> = ({
...
+
+              {/* <div className="candy-filter-title">Attributes</div>
+               {FILTER_ATTRIBUTES_MOCK.map((attr) => {
+                return (
+                  <div className="candy-filter-attribute">
+                    <span>{attr.name}</span>
+                    <Dropdown items={attr.options} onSelectItem={onFilterAttribute} placeholder={attr.placeholder} />
+                  </div>
+                );
+              })} */}
```
Comment (haVincy) [inline core/ui/src/public/Orders/index.tsx:217]: Please remove unused codes

Comment (vanminh1701) [reply core/ui/src/public/Orders/index.tsx]: This UI for future feature, if. I can't place it here, where can I place it ?

File: core/ui/src/components/Dropdown/index.tsx (hunk 6 above)
Comment (vanminh1701) [reply core/ui/src/components/Dropdown/index.tsx:38]:
```
        {currentlySelectedItem?.label ? (
          <div
            className={`candy-line-limit-1 candy-dropdown-label candy-dropdown-label--${
              isMenuOpen ? 'purple' : 'black'
            }`}
            title={currentlySelectedItem.label}
          >
            {currentlySelectedItem.label}
          </div>
        ) : (
          <div>{placeholder}</div>
        )}
        <div className="candy-dropdown-icon">{isMenuOpen ? <IconChevronUp /> : <IconChevronDown />}</div>
```
updated will be, It used 1 time, so I think we don't need to update follow your idea

Comment (haVincy) [reply core/ui/src/components/Dropdown/index.tsx:38]: alright.

**Pull request merged**